osu.Game/Graphics/Containers/ScalingContainer.cs

```diff
@@ -111,6 +118,15 @@ private void load(OsuConfigManager osuConfig)
 
             protected override void Update()
             {
+                // If the window size constitutes an aspect ratio smaller than the base size,
+                // then the DrawSizePreservingFillContainer starts descaling the game to ensure it always fits.
+                // We do not desire this behaviour when the window dimensions are small already (e.g. 852x393 on iPhone 16).
+                // The number 576 is picked arbitrarily here as a cutoff point which keeps the game scaled appropriately.
+                float minimumScalingHeight = Math.Min(576f, host.Window.Size.Width / base_aspect_ratio);
+                float windowSizeFactor = Math.Min(1f, host.Window.Size.Height / minimumScalingHeight);
+                // Easiest method to apply this compensation is by adjusting the TargetDrawSize.
+                TargetDrawSize = base_target_draw_size * windowSizeFactor;
```

Comment:
It's good to see that the scaling is applied per axis 🙂

Reply:
huh? nothing here is per axis? `windowSizeFactor` is a single float, so this is vector times float? where are you seeing "per axis"?

Reply:
I more so meant that this calculation will accurately account for the aspect ratio because the scaling factor is adjusted based on the window width against the base aspect ratio to determine an appropriate scaling height.